### ISSUE TITLE
Cannot log in

### ISSUE THREAD

**Issue by danstewart** [nextcloud/desktop#2837]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Issue

Sorry if this has been raised already but I can't find it.

I'm unable to login using the desktop client on Linux. I've tried the appimage on Manjaro and the PPA on Ubuntu 20.04 and both are the same. 

It fails to open the browser and the copy link button does nothing.

I am running the server via docker behind an nginx reverse proxy, but have tried logging into the desktop client using the IP and port and it was the same.

### Expected behaviour
Desktop client should open browser or allow copying link.

### Actual behaviour
- Client displays error "Unable to open the Browser, please copy the link to your Browser"
- Clicking "Copy link" does not put anything in clipboard

### Steps to reproduce
1. Download appimage
2. Run appimage
3. Click "Re-open in Browser" or "Copy link"

### Client configuration
Client version:

Operating system: Manjaro (Tested on Ubtunu 20.04 too)

OS language: English (UK)

Qt version used by client package (Linux only, see also Settings dialog): Not sure

Client package (From Nextcloud or distro) (Linux only): appimage from nextcloud

Installation path of client: `~/bin/Nextcloud-3.1.1-x86_64.AppImage`

### Server configuration

Nextcloud version: 20.0.5.2

Storage backend (external storage): None

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

1. Client logfile: 
https://gist.github.com/danstewart/f4a9ff155eb61bd26f6a64043e9a67b8

2. Web server error log:
```
192.168.0.1 - - [18/Jan/2021:19:58:34 +0000] "GET /status.php HTTP/1.1" 200 1649 "-" "Mozilla/5.0 (Linux) mirall/3.1.1git (build 4317) (Nextcloud)"
192.168.0.1 - - [18/Jan/2021:19:58:34 +0000] "GET /remote.php/webdav/ HTTP/1.1" 401 1646 "-" "Mozilla/5.0 (Linux) mirall/3.1.1git (build 4317) (Nextcloud)"
192.168.0.1 - - [18/Jan/2021:19:58:34 +0000] "PROPFIND /remote.php/webdav/ HTTP/1.1" 401 1640 "-" "Mozilla/5.0 (Linux) mirall/3.1.1git (build 4317) (Nextcloud)"
192.168.0.1 - - [18/Jan/2021:19:58:34 +0000] "GET /ocs/v2.php/cloud/capabilities?format=json HTTP/1.1" 200 1166 "-" "Mozilla/5.0 (Linux) mirall/3.1.1git (build 4317) (Nextcloud)"
192.168.0.1 - - [18/Jan/2021:19:58:34 +0000] "POST /index.php/login/v2 HTTP/1.1" 200 1080 "-" "manjaro (Desktop Client - Linux)"
```

3. Server logfile: nextcloud log (data/nextcloud.log):
Nothing is logged here while trying to log in

Server config: https://gist.github.com/danstewart/e474c6a89129556781ebe77db22d85de

**Comment by er-vin** [comment]
Could we have a bunch of screenshots of the issue? From your description, I'm not quite sure where the error got displayed.

**Comment by danstewart** [comment]
Hi, yes, sorry:

This is displayed after entering the URL and clicking next
![image](https://user-images.githubusercontent.com/10670565/105014213-10b68e80-5a38-11eb-87cd-67aaafa613ac.png)


Then this is displayed a few seconds later:
![image](https://user-images.githubusercontent.com/10670565/105013914-a867ad00-5a37-11eb-9868-4257178fc929.png)


Clicking "Re-open browser" shows the first error again, "Copy link" seems fine in the UI but just empties the clipboard.

Also for what it's worth, I've tried the latest daily build, release 3.0.3 and release 2.6.0 - it's the same on all of them.
I think it must be a server config issue but the lack of error message on the server makes me less sure.

**Comment by er-vin** [comment]
Yes, this looks like a server config issue which doesn't give us the right URL on the client side. Try to look into the debug logs of the client but I suspect it somehow gets an empty URL.

**Comment by danstewart** [comment]
Thanks, I'll try do some more digging tonight on the server.

**Comment by danstewart** [comment]
Set the log level to 0 and I get the following when clicking "next" after entering the URL:
> No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured

https://gist.github.com/danstewart/a485088cf2ba30e8c8728f9c549e2d79

and I get only this when clicking "Copy link"
```
{"reqId":"izv67ApgX9QxDyqRVFsp","level":0,"time":"2021-01-21T20:20:08+00:00","remoteAddr":"86.139.116.231","user":"--","app":"files_sharing","method":"POST","url":"/index.php/login/v2","message":"/appinfo/app.php is deprecated, use \\OCP\\AppFramework\\Bootstrap\\IBootstrap on the application class instead.","userAgent":"manjaro (Desktop Client - Linux)","version":"20.0.5.2"}
```

**Comment by danstewart** [comment]
I've added this to index.php
```
throw new \Exception(print_r($_SERVER));
```

and ran `curl -H "Authorization: Basic user:pass" $url` and it shows the header in the response:
```
    [HTTP_AUTHORIZATION] => Basic user:pass
```

So looks like the header is being passed through to the app correctly.

**Comment by er-vin** [comment]
By debug logs I meant on the client side. This way we'd see what is received from the server that the client tries to reach.

**Comment by danstewart** [comment]
Sorry - how do I get the client debug logs?

**Comment by er-vin** [comment]
If you use 3.1 it's just about clicking on the "Create Debug Archive" button in the settings dialog. Otherwise:
https://docs.nextcloud.com/desktop/3.0/troubleshooting.html#saving-files-directly

**Comment by danstewart** [comment]
Here you go:
https://gist.github.com/danstewart/9e73dffb17e5e5ed370ea0386467d37f

**Comment by er-vin** [comment]
This indeed looks like there's something fishy on the server side, from the logs the client is doing its job properly of contacting the right end point but then gets an empty URL as poll endpoint. I'm not quite sure why that would happen to be honest. I'll transfer this to the server repository in case they have an idea.
